packages/next-intl/src/navigation/utils.tsx

```diff
@@ -96,15 +96,16 @@ export function compileLocalizedPathname<Locales extends AllLocales, Pathname>({
   query?: Record<string, SearchParamValue>;
 }) {
   function getNamedPath(value: keyof typeof pathnames) {
-    const namedPath = pathnames[value];
+    let namedPath = pathnames[value];
     if (!namedPath) {
-      throw new Error(
-        process.env.NODE_ENV !== 'production'
-          ? `No route found for "${value}". Available routes: ${Object.keys(
-              pathnames
-            ).join(', ')}`
-          : undefined
-      );
+      namedPath = value;
+      if (process.env.NODE_ENV !== 'production') {
+        console.warn(
```

Comment:
Throwing an error is definitely too much, thanks for raising this point. A warning is better, but I'm wondering if we should just process the `pathname` as-is in this case, without any message to the user? Wouldn't that be necessary when you pass unknown strings to the navigation APIs?

Reply:
Agreed.

1. If strict, we want to warn user of mistake
2. If allow any URL, we don't want to break the app.

As I mentioned in the above comment, `console.warn` isn't perfect, but I think it satisfies both use-cases.

Reply:
I'm just worried that even `console.warn` creates unnecessary noise if the user intentionally passes an unknown pathname. E.g. in an app like yours, wouldn't it be annoying to see a warning every time you're redirected to `/api/v1/billing/checkout` during development?

I think at this point it's quite safe to assume that most developers use TypeScript, so I think we can rely a bit more on the types for catching errors. If you want to globally opt-in to allowing unknown routes, you can modify the types for `pathnames` as shown above and if you want more granular control, you can still `// @ts-expect-error` for individual cases.